sdk/Sdk.Generators/Extensions/ListExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    internal static class ListExtensions
+    {
+        /// <summary>
+        /// If the enumerable passed in is already a list, it is returned, otherwise ToList() is invoked.
+        /// </summary>
+        /// <typeparam name="T">The type of element in the list.</typeparam>
+        /// <param name="source">The enumerable to return as a list.</param>
+        internal static List<T> AsList<T>(this IEnumerable<T>? source)
```

Comment:
Can you expand on why you feel this extension is needed in this scenario over simply using `ToList`?

The default behavior will trigger an allocation, but this is only being avoided for one of the methods you have currently using this. Is there something specific you're trying to address with this? Did you observe significant issues/impact when using `ToList`?

Reply:
Synced offline. Pushed an update - Removed the extension method.